eng/scripts/Generate-Patch.ps1

```diff
@@ -0,0 +1,109 @@
+# 5. BranchName - The name of the remote branch where the changes will be pushed. We push the changes to the 'upstream remote'. This is not a required parameter.
+#                 In case the argument is not provided the branch name is release/{ArtifactName}.
+#
+# Example:  .\eng\scripts\Generate-Patch.ps1 -ArtifactName azure-mixedreality-remoterendering -ServiceDirectory remoterendering -ReleaseVersion 1.0.0 -PatchVersion 1.0.1
+# This creates a remote branch "release/azure-mixedreality-remoterendering" with all the necessary changes.
+
+param(
+  [Parameter(Mandatory=$true)][string]$ArtifactName,
+  [Parameter(Mandatory=$true)][string]$ServiceDirectoryName,
+  [Parameter(Mandatory=$true)][string]$ReleaseVersion,
+  [Parameter(Mandatory=$true)][string]$PatchVersion,
+  [Parameter(Mandatory=$false)][string]$BranchName
+)
+
+function TestPathThrow($Path) {
+  if(!(Test-Path $Path)) {
+   LogError "$Path not found"
+   exit
+  }
+}
+
+Write-Information "PS Script Root is: $PSScriptRoot"
+Write-Information "ArtifactName is: $ArtifactName"
+Write-Information "ReleaseVersion is: $ReleaseVersion"
+Write-Information "ServiceDirectoryName is: $ServiceDirectoryName"
+Write-Information "PatchVersion is: $PatchVersion"
+
+if($null -eq $BranchName) {
+  $BranchName = "release/$artifactName.ToLower()"
+}
+Write-Information "BranchName is: $branchName"
+
+$RepoRoot = Resolve-Path "${PSScriptRoot}..\..\.."
+$EngDir = Join-Path $RepoRoot "eng"
+$EngCommonScriptsDir = Join-Path $EngDir "common" "scripts"
+$SdkDirPath = Join-Path $RepoRoot "sdk"
+$ServiceDirPath = Join-Path $SdkDirPath $ServiceDirectoryName
+$ArtifactDirPath = Join-Path $ServiceDirPath $ArtifactName
+$ArtifactPomFile = Join-Path $ArtifactDirPath "pom.xml"
+$ReleaseTag = -join($ArtifactName, "_", $ReleaseVersion)
+$TestResourcesFilePath = Join-Path $ServiceDirPath "test-resources.json"
+$GroupId = "com.azure"
+
+TestPathThrow -Path $ArtifactDirPath
+TestPathThrow -Path $EngDir
+TestPathThrow -Path $EngCommonScriptsDir
+TestPathThrow -Path $ArtifactDirPath
+TestPathThrow -Path $ArtifactPomFile
+TestPathThrow -Path $SdkDirPath
+
+## Creating a new branch
+git checkout -b $BranchName upstream/main
+
+try {
+  ## Hard reseting it to the contents of the release tag.
+  ## Fetching all the tags from the upstream branch
+  $CmdOutput = git fetch upstream --tags
+  $cmdOutput = git restore --source $ReleaseTag -W -S $ArtifactDirPath
+  if($LASTEXITCODE -ne 0) {
+    LogError "Could not restore the changes for release tag $ReleaseTag"
+    exit
+  }
+
+  if(Test-Path $TestResourcesFilePath) {
+    $cmdOutput = git restore --source $ReleaseTag -W -S $TestResourcesFilePath
+  }
+
+  ## Commit these changes.
+  # git commit -a -m "Reset changes to the patch version."
+  
+  ## Create the patch release
+  python "$EngDir/versioning/set_versions.py" --bt client --new-version $PatchVersion --ar $ArtifactName --gi $GroupId
```

Comment:
Any reason this doesn't use `Join-Path` to ensure the proper usage of `/` vs `\`

Reply:
I came across Join-Path after I wrote this :). I will update this for consistency.